pylzy/lzy/api/v1/remote/workflow_service_client.py

```diff
@@ -138,9 +148,29 @@ async def start_workflow(self, name: str, storage: Storage, storage_name: str) -
         else:
             raise ValueError(f"Invalid storage credentials type {type(storage.credentials)}")
 
-        res: StartWorkflowResponse = await self.__stub.StartWorkflow(
-            StartWorkflowRequest(workflowName=name, snapshotStorage=s, storageName=storage_name)
-        )
+        try:
+            res: StartWorkflowResponse = await self.__stub.StartWorkflow(
+                StartWorkflowRequest(workflowName=name, snapshotStorage=s, storageName=storage_name)
+            )
+        except AioRpcError as e:
+            data = e.trailing_metadata().get_all("x-supported-client-versions")
```

Comment:
I think you should rewrite it as

```
except AioRpcError as e:
    lzy.urtils.grpc.raise_for_version(e)  # incapsulates logic version check and raises if it is a case,
    raises. Also, it allows to reuse this logic at other places. 
```

Also maybe we should think about setting exceptionhook in this kind of exception except of try/excepting everywhere.

Reply:
I made a decorator to make it easier to override errors here.